app/models/broadcast.rb

```diff
@@ -2,7 +2,7 @@ class Broadcast < ApplicationRecord
   VALID_BANNER_STYLES = %w[default brand success warning error].freeze
   resourcify
 
-  has_many :notifications, as: :notifiable, inverse_of: :notifiable
+  has_many :notifications, as: :notifiable, inverse_of: :notifiable, dependent: :destroy
```

Comment:
The only thing I wonder is whether we should alert admins right before they delete a `Broadcast` and let them know [in the delete method of the controller](https://github.com/forem/forem/blob/master/app/controllers/admin/broadcasts_controller.rb#L49-L59) that the action they're about to take will also delete all associated `Notifications`? I can totally see admins playing with this feature and not necessarily knowing that deleting one model destroys the others 🤷

Reply:
We can replace this with `:restrict_with_error` to make sure notifications are explicitly destroyed before ? Is it possible to destroy notifications from the admin panel?

Reply:
@rhymes AFAIK, you can't destroy notifications from the admin panel (?), but I do like the idea of us either forcing the admin to explicitly destroy them (somehow) or alerting them that they will be destroyed as a result of the broadcast being deleted.

Reply:
@vaidehijoshi should we make an issue for option 2? Maybe "destroy broadcast" can have a confirmation step where the user is notified that N notifications are going to be deleted and then they can decide whether to confirm or bail?